### PR TITLE
#95 ignore white spaces before and after quotes/fields/values

### LINKED ISSUE #95: Additional spaces cause CSV parsing to fail
Steps to replicate: 

let failingCsv = """
"a", "b"
"A","M"
"""
let csv: CSV = try CSV(string: value, delimiter: ",")

If you try to remover the space, it parses the CSV without problems: 
```
"a","b"
"A","M"
```

### PR DESCRIPTION
**Pull request by ondrej-kvasnovsky**
Fixes #95

### CODE REVIEW

File: SwiftCSVTests/CSVTests.swift [hunk 1]
```diff
@@ -122,4 +122,14 @@ class CSVTests: XCTestCase {
             XCTAssert(error is CSVParseError)
         }
     }
+  
+    func testInit_ParseFileWithQuotesAndSpaces() {
```
Comment (DivineDominion) [inline SwiftCSVTests/CSVTests.swift:126]: Since the implementation checks for `isWhitespace`, could you add similar-looking test cases for other whitespace characters, too?

**Review by DivineDominion — CHANGES_REQUESTED** [review]
Looks good so far, thanks for adding the test case and providing a fix!

Would be great if you could add unit tests for other whitespace characters that should be skipped, too. Tabs come to mind; more exotic spaces like the non-breaking space (⌥+space on some keyboard layouts) would be optional "stretch goals" :)

**Comment by ondrej-kvasnovsky** [comment]
> Looks good so far, thanks for adding the test case and providing a fix!
> 
> Would be great if you could add unit tests for other whitespace characters that should be skipped, too. Tabs come to mind; more exotic spaces like the non-breaking space (⌥+space on some keyboard layouts) would be optional "stretch goals" :)

Ok ok. Added more test data (using https://developer.apple.com/documentation/swift/character/3127019-iswhitespace). That that is pretty much the intent of this PR.

**Review by DivineDominion — APPROVED** [review]
Thanks!

**Comment by DivineDominion** [comment]
We've invited you to join the SwiftCSV GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/SwiftCSV/SwiftCSV/blob/master/CONTRIBUTING.md).
